contracts/bond/DaoAccessControl.sol

```diff
@@ -0,0 +1,299 @@
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/structs/EnumerableSetUpgradeable.sol";
+
+// TODO doco
+
+// TODO not useful, needs the roles setup in child
+abstract contract DaoAccessControl is Initializable, ContextUpgradeable {
+    using EnumerableSetUpgradeable for EnumerableSetUpgradeable.Bytes32Set;
+
+    struct Role {
+        EnumerableSetUpgradeable.Bytes32Set adminRoles;
+        mapping(address => bool) members;
+    }
+
+    // Roles that apply in DAOs individually
+    mapping(uint256 => mapping(bytes32 => Role)) private _daoRoles;
+
+    // Roles that apply across all DAOs
+    mapping(bytes32 => Role) private _globalRoles;
+
+    event AddDoaRoleAdmin(uint256 daoId, bytes32 role, bytes32 adminRole);
+    event AddGlobalRoleAdmin(bytes32 role, bytes32 adminRole);
+    event GrantDaoRole(uint256 daoId, bytes32 role, address account);
```

Comment:
Do we want to add `address` grantee (`msg.sender`) to this event so we can log who granted who permissions?

Reply:
yes, yes we do 👍

Reply:
(while not needed for the eventLog, it'll be required for graphQL, which lacks the `from` address in the transaction)

Reply:
This will be done later & applied to all events in this repo https://github.com/windranger-io/windranger-treasury/issues/245

Reply:
FWIW I think the subgraph Transaction Class does have the `from` field but no concept of `msg.sender` so in the case `msg.sender` != `tx.origin` or `from` necessitates including msg.sender from the event

Reply:
Hmm.... that could make things tricky 🤔 

Whenever a contract does a `call` we'll loose the details of the contract performing the op (which would be missing from the access list).

Although maybe that's only relevant in the `revert` conditions.